### ISSUE TITLE
Error "'x' must be an array of at least two dimensions" when passing a data.table instead of data.frame

### ISSUE THREAD

**Issue by CyrilLagger** [krassowski/complex-upset#102]
Hi, 

I just noticed that passing a data.table instead of a data.frame to `upset()` produce this error:

```R
Error in colSums(inclusive_intersection) : 
  'x' must be an array of at least two dimensions
```
It can be reproduced directly from the movie example from your vignette:
```R
library(data.table)
upset(data = data.table::setDT(movies), intersect = genres)
```

I didn't check the details, but it probably comes from small syntax differences between data.frame and data.table ([see e.g. here](https://cran.r-project.org/web/packages/data.table/vignettes/datatable-faq.html#SmallerDiffs)).

I wouldn't consider this a bug as `?upset` clearly states that `data` has to be a data.frame. I actually could solve the problem in my case by calling `upset(data = data.table::setDF(my_data_table), ...)`. I thought I would mention it anyway as it took me some time to realize what was the problem (since in most cases function designed for data.frame also work for data.table and I was first looking in other directions). 

Best and thanks for this package!

**Context**

ComplexUpset version: 1.1.0

<details>
<summary>R session information</summary>

```R
R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17763)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252  LC_CTYPE=English_United Kingdom.1252   
[3] LC_MONETARY=English_United Kingdom.1252 LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] data.table_1.13.6  ggplot2_3.3.3      ComplexUpset_1.1.0
```

</details>

**Comment by krassowski** [comment]
Thanks for bringing this up, it will certainly be very useful for others to find this issue based on the error message.

It makes sense to have a simple utility to automatically convert data tables to data frames I think. Would it be what you would expect? Complex-upset is already doing that if it gets a `tibble`s, here:

https://github.com/krassowski/complex-upset/blob/7c2e1e46dff720dda183b596159f5031601aef4c/R/data.R#L344-L346

and there is a related test here:

https://github.com/krassowski/complex-upset/blob/df90dc9f8295d2c0d046ce94a90b661c28394e6a/tests/testthat/test-data.R#L351-L373

What I would very much like to avoid is to have a dependency on data.table; if it is possible to convert without adding a dependency (using `recommends` for tests is fine) I would welcome pull requests.

**Comment by CyrilLagger** [comment]
Thanks for the reply. Actually, I just checked that `as.data.frame(my_data_table)` do also convert a data.table to a data.frame and would actually make more sense than using `data.table::setDF(my_data_table)` as proposed in my first post. The point of `setDF` is to change the object by reference without any copy, which means that after calling `upset(data.table::setDF(my_data_table), ...)`, `my_data_table` will be a data.frame in the rest of the code, which is probably not what a user would want. 

So I think it should be sufficient to modify your line 344-346 by something as follows:
```R
 if ('tbl' %in% class(data) | 'data.table` %in% class(data)) { 
     data = as.data.frame(data) 
 } 
```
That will create a copy of `data` but the original object won't be modified, and there is no dependency on data.table.

**Comment by krassowski** [comment]
I am sorry, I must have missed your reply earlier. The solution you propose looks very nice; would you like to contribute it by opening a pull request?

**Comment by krassowski** [comment]
Closing as this was solved by #105 and released in 1.2.0. Thank you for reporting and help in fixing it!
